### PR TITLE
Support libraries/gradle updates, update to work with Android Studio 2.0

### PR DESCRIPTION
**Pull request by something15525**
Update support libs, gradle version. Fix layout bug with new design library.

### CODE REVIEW

**Comment by something15525** [comment]
@JakeWharton looks like the build failed because of a missing 23.0.3 build tools.

File: gradle/wrapper/gradle-wrapper.properties [hunk 1]
```diff
@@ -1,6 +1,6 @@
...
-distributionPath=wrapper/dists
-zipStoreBase=GRADLE_USER_HOME
-zipStorePath=wrapper/dists
-distributionUrl=https\://services.gradle.org/distributions/gradle-2.9-all.zip
+#Tue Apr 19 10:16:35 PDT 2016
+distributionBase=GRADLE_USER_HOME
+distributionPath=wrapper/dists
+zipStoreBase=GRADLE_USER_HOME
+zipStorePath=wrapper/dists
+distributionUrl=https\://services.gradle.org/distributions/gradle-2.10-all.zip
```
Comment (JakeWharton) [inline gradle/wrapper/gradle-wrapper.properties]: 2.12 is latest

File: build.gradle [hunk 2]
```diff
@@ -65,7 +65,7 @@ def preDexEnabled = "true".equals(System.getProperty("pre-dex", "true"))
 
 android {
   compileSdkVersion 23
-  buildToolsVersion '23.0.2'
+  buildToolsVersion '23.0.3'
```
Comment (JakeWharton) [inline build.gradle]: You need to update `.travis.yml` if you're changing build tools version.

**Comment by something15525** [comment]
Odd, it seems to have failed on a lint error:
`/home/travis/build/JakeWharton/u2020/src/androidTestInternal/java/com/jakewharton/u2020/U2020TestRunner.java:26: Error: Missing permissions required by KeyguardLock.disableKeyguard: android.permission.DISABLE_KEYGUARD [MissingPermission]
    keyguard.newKeyguardLock(name).disableKeyguard();
    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
Applies to variants: internalRelease`

Any ideas on why this would be happening?

**Comment by JakeWharton** [comment]
Lint bug. We suppressed it on the method internally.

**Comment by something15525** [comment]
So if we merge this into master this won't be an issue? Just trying to get clarification on next steps to get this PR merged. Thanks!

**Comment by JakeWharton** [comment]
Yep. Just annotate that method with @SuppressLint and the name of the lint
check.

**Comment by JakeWharton** [comment]
Thanks!

**Pull request merged**